ogr/ogrsf_frmts/generic/ogrlayer.cpp

```diff
@@ -669,6 +669,44 @@ OGRErr OGR_L_CreateFeature( OGRLayerH hLayer, OGRFeatureH hFeat )
     return OGRLayer::FromHandle(hLayer)->CreateFeature( OGRFeature::FromHandle(hFeat) );
 }
 
+/************************************************************************/
+/*                           UpsertFeature()                           */
+/************************************************************************/
+
+OGRErr OGRLayer::UpsertFeature( OGRFeature *poFeature )
```

Comment:
this method would need to be documented in ogr/ogrsf_frmts/ogrsf_frmts.dox
The doc of CreateFeature() and UpdateFeature() (in that .dox file) should likely link to this new method

Reply:
Do you have preferences on how things are linked? Do you just want links in OGRLayer::CreateFeature and OGRLayer::SetFeature (ignoring the other variations like ICreateFeature and OGR_L_SetFeature)?

For example:

In create: To create a feature, but set it if it exists see OGRLayer::UpsertFeature()
In set: To set a feature, but create it if it doesn't exist see OGRLayer::UpsertFeature()

Reply:
> Do you just want links in OGRLayer::CreateFeature and OGRLayer::SetFeature 

yes

> For example:
> In create: To create a feature, but set it if it exists see OGRLayer::UpsertFeature()
> In set: To set a feature, but create it if it doesn't exist see OGRLayer::UpsertFeature()

sounds good!